src/libsyntax/parse/lexer/mod.rs

```diff
@@ -43,16 +43,16 @@ pub struct UnmatchedBrace {
 }
 
 pub struct StringReader<'a> {
-    pub sess: &'a ParseSess,
+    crate sess: &'a ParseSess,
     /// The absolute offset within the source_map of the next character to read
-    pub next_pos: BytePos,
+    crate next_pos: BytePos,
```

Comment:
A bikeshed question: What does `crate` keyword do here? I cannot find the answer in [rust reference book](https://doc.rust-lang.org/nightly/reference/visibility-and-privacy.html).

Reply:
It's an unstable syntax for `pub(crate)` visibility

Reply:
Thank you.